### DISCUSSION TITLE
URI link from XDrip to Atlas/Mongodb

### DISCUSSION THREAD

**Discussion by DavidASmithPE** [NightscoutFoundation/xDrip#2205] [Q&A]
Several months ago I tried to set up nightscout using the "Classic DIY Installation". I think that I did the installation correctly, but I can't upload to it. I logged into Heroku and looked at the database and extracted the URL and installed it in XDrip but to no avail. I am attaching some screen shots to show what I have and am getting. My wife's phone is a Galaxy A34 and we are running Dexcom G6.
![Screenshot_20220624-190123_xDrip+](https://user-images.githubusercontent.com/71335540/177013630-0312be16-ad31-4a34-8251-79a6c2701802.jpg)
![Screenshot_20220624-190211_xDrip+](https://user-images.githubusercontent.com/71335540/177013632-ac55bf33-9b1a-49af-8ef9-bb1c54af8e80.jpg)
![Screenshot_20220614-222339_xDrip+](https://user-images.githubusercontent.com/71335540/177013634-fe78df80-bb80-4d57-890b-e3a65dd7bfdc.jpg)

**Comment by Navid200** [comment]
@psonnera

**Reply by DavidASmithPE** [reply]
??

**Comment by jonprogers** [comment]
I think you are trying to use your mongo db uri to upload to nightscout with the REST api. The URI you are trying to write is for the direct mongo db upload...

I think your Base URL on the Nightscout Sync (REST-API) settings page should be set to https://<password>@<site>.herokuapp.com/api/v1 - or at least that is what mine is set to!

**Reply by jonprogers** [reply]
sorry, the setting didn't paste in properly - https://PASSWORD@SITE-PREFIX.herokuapp.com/api/v1

**Reply by DavidASmithPE** [reply]
by SITE-PREFIX do you mean the database name; in my case CeciliaCGMIC?

**Reply by jonprogers** [reply]
no, the address you use to view your nightscout site that ends in .herokuapp.com - assuming you have your nightscout site hosted on heroku...

**Reply by jonprogers** [reply]
and the PASSWORD is whatever you have set as API_SECRET in your heroku config vars

**Reply by DavidASmithPE** [reply]
I may have a PW issue that I can work on, but here is the screen shot now
![Screenshot_20220702-160243_xDrip+](https://user-images.githubusercontent.com/71335540/177014861-f430a547-722b-459e-84c7-09a2ba2bf9b0.jpg)
.

**Reply by jonprogers** [reply]
see the link from @psonnera - he knows what he's talking about, I'm just guessing!!

**Reply by DavidASmithPE** [reply]
OK, I think that is what I had. Here is a clip that I just got from Heroku. So this should be the SITE-PREFIX?
![InHeroku](https://user-images.githubusercontent.com/71335540/177015182-96e937d2-80bb-4662-8962-6e3a82333f55.jpg)

**Reply by jonprogers** [reply]
yes, looks like your SITE-PREFIX is ced... followed by the bit under the blue blob!

**Reply by jonprogers** [reply]
and if you click on your prefix, then go to the settings tab and then click the 'Reveal Config Vars' button you will see your list of configuration variables. There should be one called API_SECRET, you password is whatever that variable is set to.

good luck!

**Reply by psonnera** [reply]
@jonprogers you got it all right, I just missed the main thread when answering. @DavidASmithPE yes this Heroku app name is your site prefix in the example above. As for the password it's your API_SECRET: you'll find it in the Heroku variables of your site. https://nightscout.github.io/nightscout/new_user/#editing-config-vars-in-heroku

**Reply by DavidASmithPE** [reply]
Well, I think I am getting closer. I think the PW and site name are OK; however, there must be something else.
![Screenshot_uploadpg_xDrip+](https://user-images.githubusercontent.com/71335540/177016599-c1620d14-845e-4e5c-bba0-8fd590c700d6.jpg)
![Screenshot_20220702-171211_xDrip+](https://user-images.githubusercontent.com/71335540/177016600-82072cd9-42f3-446c-b1ce-9ce6ab4e4b66.jpg)

**Reply by psonnera** [reply]
Does the Nightacout page open when you browse it? Did you create a new profile and set the correct time zone in it?

**Reply by DavidASmithPE** [reply]
This is something that I needed to do. When I opened the Nightscout page there was nothing filled in. I put in the name of the SITE-PREFIX and the time and saved it. Data looked like it started but it is still not working. Attached is the XDrip status page and a clip of the Nightscout page. I think you are getting me closer to having it work. Thanks for your continuing help. Sorry the WEB page clip is so small.
![Screenshot_20220703-030727_xDrip+](https://user-images.githubusercontent.com/71335540/177029670-ef281e1c-acaa-4f47-b8e2-43a4a0429d6d.jpg)
![Nightscout clip](https://user-images.githubusercontent.com/71335540/177029672-fb792fa2-ecd9-4107-833b-6930f9af8cff.jpg)

**Reply by jonprogers** [reply]
OK, your profile page looks OK to me, were you able to save the record OK. You don't show the bottom of the page, that will show your authorisation status - you also use your API_SECRET to authorise the profile editor so it's a good check on that, mine shows 'Admin authorized'.

What is your main page showing (at ceda1...herokuapp.com) - from memory, with no CGM readings in the database, that should show the correct time and a mainly blank space for a graph together with a few other bits depending on your settings.

You could also try going to ceda1...herokuapp.com/api/v1/status.json This should give you a page of json text data that shows your site status and settings, it does it through the same api interface (the v1 API) that xDrip uses to upload data but as you are requesting data rather than uploading it I don't think you need to authenticate with the API_SECRET - at least I don't!
Early lines in my data show things like status: ok, apiEnabled: true and the Nightscout version number which it would be good that you confirm seeing...

Hopefully if you can confirm what your nightscout site is generating it will narrow things down a bit...

**Reply by jonprogers** [reply]
Also could you just double check your base url in xDrip - from the screenshot of your profile setting page your site prefix ends in the letters 'bg' but in the xDrip setting page screenshot where you obscured the prefix it looks like the last two letters could be 'db' - can't be sure as you blanked it out...
![image](https://user-images.githubusercontent.com/19576623/177031990-d253a804-f925-4493-a5a3-a168f82880c2.png)
![image](https://user-images.githubusercontent.com/19576623/177032014-62c3ae09-72b0-4e26-9c7d-97a34ad7c016.png)

**Reply by DavidASmithPE** [reply]
Jon, This is fantastic! Thanks so much and picking off my typing error was super. I think I am close to getting it right, but I still have a few items I would like to fix. I have the data uploading to NightScout:

![NS OP](https://user-images.githubusercontent.com/71335540/177049582-15e9e406-cab9-48fa-84ec-3fa6daa5e714.jpg)

, but I still have an error on the system page #3:

![Screenshot_20220703-121524_xDrip+Sys#3](https://user-images.githubusercontent.com/71335540/177049635-6fe6bc9d-51b2-47c2-9ec3-5b564af8e578.jpg)

and on the NS graph it shows the insulin usage but does not define it as short acting or long acting. The 4 units is Humalog and the 19U is Lantus and is shown that way in XDrip. Anyway to fix that? Cecilia does not us a pump.

The json data is:

![json](https://user-images.githubusercontent.com/71335540/177049783-a4d82662-22c7-4084-b44c-b81e54c8132f.jpg)

SYS pg#1 is:

![Screenshot_20220703-121550_xDrip+Sys#1](https://user-images.githubusercontent.com/71335540/177049821-94282dbd-4412-4799-bbcc-a611200ac44b.jpg)
Pg#2 is

![Screenshot_20220703-121535_xDrip+Sys#2](https://user-images.githubusercontent.com/71335540/177049835-5e45d89f-6478-48c2-8bca-7ebcf6a8daac.jpg)

 Cloud NightScout screen is:

![Screenshot_20220703-121629_xDrip+Cloud NS](https://user-images.githubusercontent.com/71335540/177050106-e3b3c466-6474-415d-b6ae-f2401063bc61.jpg)

And the Extra selection is:

![Screenshot_20220703-121651_xDrip+Extra](https://user-images.githubusercontent.com/71335540/177049909-c4806651-2362-4126-ac98-4fc3b4ff1898.jpg)

![Screenshot_20220703-121706_xDrip+Extra bot](https://user-images.githubusercontent.com/71335540/177050470-d4d77d37-9e63-4347-bc1a-e30f01a5345d.jpg)

I hope that gives enough information to pickoff what I need to do next.
In any case, THANK YOU SO MUCH!

**Reply by jonprogers** [reply]
1) great you have data!!
2) the error listed is over 2 hours prior to your screenshot - I believe this always shows the last error and the time it happened so all is good
3) I am a pump user so have no experience of multi insulin functions, I hope someone else can help you here as I can't
4) status.json page was to try to see if it was a nightscout or xDrip problem - not needed now we know it was an xDrip setting error and you now have data!

**Reply by psonnera** [reply]
@DavidASmithPE xDrip+ won't send injections type information and treatments will only be insulin. This is matching Nightscout behaviour as basal is expressed in your profile https://nightscout.github.io/nightscout/profile_editor/ and not in careportal https://nightscout.github.io/nightscout/discover/#add-a-treatment-careportal

**Comment by psonnera** [comment]
The string you're using is your MongoDB URI, you cannot use it in xDrip+. Log into Heroku and find your Nightscout app name, this is the one you need to use. https://nightscout.github.io/uploader/setup/#xdrip as said above, Mongo Direct is now deprecated and shouldn't be used unless you host your own MongoDB database, not hosted from MongoDB Atlas.
